src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -171,13 +173,59 @@ export class BeneficiariesService {
 		try {
 			const user = await this.userService.ipUserInfo(req);
 
+			const variables: any = {};
+
+			let filterQueryArray = [];
+			let paramsQueryArray = [];
+
+			if (body.search && body.search !== '') {
+				 let first_name = body.search.split(" ")[0]
+				 let last_name = body.search.split(" ") ? body.search.split(" ")[1] : ""
+
+			if (last_name?.lasy_namelength > 0) {
+					filterQueryArray.push(`{_or: [
+				{ first_name: { _ilike: "%${first_name}%" } }
+				{ last_name: { _ilike: "%${last_name}%" } }
+				 ]} `);
+				} else {
+
+					filterQueryArray.push(`{_or: [
+				{ first_name: { _ilike: "%${first_name}%" } }
+				 ]} `);
+				}
+			}
+
+			if (body.hasOwnProperty('district') && body.district.length) {
+				paramsQueryArray.push('$district: [String!]');
+				filterQueryArray.push('{district: { _in: $district }}');
+				variables.district = body.district;
+			}
+
+			if (body.hasOwnProperty('block') && body.block.length) {
+				paramsQueryArray.push('$block: [String!]');
+				filterQueryArray.push('{block: { _in: $block }}');
+				variables.block = body.block;
+			}
+
+			if (body.facilitator && body.facilitator.length > 0) {
+				filterQueryArray.push(
+					`{program_beneficiaries: {facilitator_id:{_in: ${JSON.stringify(
+						body.facilitator,
+					)}}}}`,
+				);
+			}
+
+			let filterQuery = '{ _and: [' + filterQueryArray.join(',') + '] }';
+
+			let paramsQuery = '';
+			if (paramsQueryArray.length) {
+				paramsQuery = '(' + paramsQueryArray.join(',') + ')';
+			}
+			let sortQuery = `{ created_at: desc }`;
+
 			const data = {
-				query: `query MyQuery {
-					users(where: {
-						_and: [
-							{ program_beneficiaries: { facilitator_user: { program_faciltators: { parent_ip: { _eq: "${user?.data?.program_users[0]?.organisation_id}" } } } } }
-						]
-					}){
+				query: `query MyQuery ${paramsQuery} {
+					users(where:${filterQuery}, order_by: ${sortQuery}){
```

Comment:
Why did you removed this condition. Please again add this:
`{ program_beneficiaries: { facilitator_user: { program_faciltators: { parent_ip: { _eq: "${user?.data?.program_users[0]?.organisation_id}" } } } } }`

Reply:
resolved comment